jenkins/run-unittests

```diff
@@ -5,9 +5,10 @@
 # NOTE:  Before invoking this script, the various bits of Python setup should
 #        have been performed, e.g., via `source jenkins/python-setup.sh`.
 
-prog="$(basename "${0}")"
 progdir="$(realpath -e $(dirname "${0}")/..)"
 
+source ${progdir}/python-cache-cleanup.sh
+
```

Comment:
I had thought about doing that, but it seemed to mush together two separate concerns: pip install of required modules and cleanly removing cached python byte-code files.

We could make it a separate step, but then we'll probably want to refactor these files further to be able to express what we want in different contexts.  By moving this into one file we split the difference and avoid solve the core problem of hung CI jobs.

Reply:
But, this isn't being "moved into one file" -- it's being moved and duplicated into _two_ files.

The Jenkins pipeline (last I looked at it) invokes three scripts; rather than do this in each of the latter two, why not do it once in the first?

I think you are equating "setup" with the Python concept of installing requirements; however, if we instead construe "setup" as "prepare the environment for the tests", then the `python-setup.py` script is exactly the place for this.